### ISSUE TITLE
Error message "Unknown action: url" when clicking repository (URL) in branch overview

### ISSUE THREAD

**Issue by kjetil-kilhavn** [larshp/abapGit#2566]
When the repository URL is clicked in the main page a browser session is opened for the URL.
When clicking the same in branch overview error message "Unknown action: url" is issued.

**Comment by jfilak** [comment]
Confirmed on the latest version in SAP Java GUI.

**Comment by kjetil-kilhavn** [comment]
I have fixed it locally, but haven't figured out how to push that up here. I'm still a beginner in Git and have never contributed to any open source projects on Github. Will ask for some assistance on slack.

**Comment by jfilak** [comment]
Can we chat on Slack - https://abapgit.slack.com/messages/C4KRH6CAJ ?

**Comment by kjetil-kilhavn** [comment]
Sent you a message on slack - is it slack live chat you mean? (I don't have my password available so I can't use the link.)

**Comment by jfilak** [comment]
But the steps are the following:
- fork this repository
- copy HTTPS url of your forked repository
- start zabapgit
- add new online repository for the package $abapgit
- use the HTTPS url
- create new branch
- commit your changes
- go back to github and open pull reques - there should be yellow bar with the green button "New Pull Request"

**Comment by sbcgua** [comment]
regression after #2540. The fix if OK, but I'll have a look how to do it without code duplication

**Comment by filak-sap** [comment]
@sbcgua  If only ABAP supported functional programming or anonymous classes like Java/C#, you would just created an internal table of (action, handler) and pass it to the parent - but in ABAP you need an interface and a class and that's too complicated and clumsy.

I am looking forward to your solution.

Maybe I would not allow pages to overwrite `METHOD zif_abapgit_gui_page~on_event.` but I would create a special method (I do not remember the name of this pattern - "template"?) with an  abap_bool return value signalizing the event was handled. Pages in the need to handle events on their would overwrite this method which would be called from `zif_abapgit_gui_page~on_event.`.

**Comment by sbcgua** [comment]
@filak-sap Exactly :) lambdas would do well.

`on_event`. I was thinking toward some kind of middlewares. Where router is the last in the stack and page and it's components are before. Actually like `router->on_event` does but in a more universal way.

Didn't think about it a lot yet. But seems that there will be a problem with registering handlers. Now all pages subclass `zcl_page`. They may have `on_event` ... and also `zcl_page` has ... so don't forget to call super ? However, the pages can also act as kind of High Order Components like [here](https://github.com/sbcgua/abap_gui_boilerplate/blob/master/src/zguibp_example.prog.abap#L27) and [here](https://github.com/larshp/abapGit/blob/master/src/ui/zcl_abapgit_gui_page_main.clas.abap#L152).

So how will GUI know which instance is able to handle ? A page or a subcomponent should somehow let GUI know that it wants to be added to event stack. Either via parameter in `render` (ref to gui instance) or during construction ... and here the though stops for the moment :)
